sklearn/preprocessing/data.py

```diff
@@ -788,7 +788,7 @@ def transform(self, X, y=None):
             warnings.warn(DEPRECATION_MSG_1D, DeprecationWarning)
 
         if sparse.issparse(X):
-            if X.shape[0] == 1:
+            if X.shape[1] != self.scale_.shape[0]:
```

Comment:
I think this is not the right fix, there should not be a switch at all on the shape. We should always be `inplace_column_scale` in the `transform` and `inverse_transform` method of the scaler estimators (`MaxAbsScaler` and `RobustScaler`).